### DISCUSSION TITLE
Eglot breaks on non-conforming lsp server

### DISCUSSION THREAD

**Discussion by benbellick** [joaotavora/eglot#1468] [General]
Been a happy user of `eglot` for a long time, nice work! There is one difficulty I encounter which is that when using the `ocaml-lsp` for working with `ReasonML` code, the lsp spits out some JSON with a "bad" character ([see the linked issue](https://github.com/reasonml/reason-react/issues/853)). This causes `eglot` to freeze up, and is only resolved once I restart the server. 

Thus, my development experience is that I am frequently restarting the lsp. I recently tried out `lsp-mode` with the same tech stack. It does report the same issue due to the bad character, but it continues onward gracefully. 

Is there the ability to do the same in `eglot`? Thanks!

**Comment by joaotavora** [comment]
Not sure this is exactly a bug on the client side, but _if_ it is some kind of limitation it's more of a `jsonrpc.el` limitation than an Eglot one.  So @svaante, who is `jsonrpc.el` maintainer now, is the right person to ask.  I recommend you make a bug report to bug-gnu-emacs@gnu.org.  I also heartily recommend that in this bug report a complete error recipe is provided so that we have a change to see the bug in action and verify what, if anything, can be done on the client's side.

For example, it would be really good to characterize what "freeze up" means.  Presumably it's some kind dysfunction, but a "freeze" would mean it would be impossible or at least difficult to restart the server (and apparently it isn't). It would also be good to specify exactly what method of "restarting the server" you are using?  Is it `M-x eglot-reconnect`?
